### PR TITLE
Cleaned up ItemCatalogue UI and linked to backend

### PR DESCRIPTION
**Pull request by Saagar3690**
### **Changes**
1. Updated Item Catalogue UI to use the MapMenuView component to display items
2. Connected to backend, so items now display for a bot given by bot id (passed through props)
3. Modified MapMenuView to have an additional optional prop called clickable, which indicates if each item card can be pressed (this is for when I implement navigation to payment screen)
4. ItemCatalogueService provides two async functions with requests to backend
   - GET request to bots/bot for a bot given by bot id
   - PUT request to addItem with body botId, itemId, and quantity (this will be used for modifying inventory which I haven't done yet)

### **In Action**
RPReplay_Final1608277212
iOS
IMG_5424
Android
Screen Shot 2020-12-17 at 11 39 34 PM

### **TODO (in order of priority)**
1. Navigate to Payment screen after clicking an item
2. Send request to backend for payment screen to modify inventory
3. Sale statistics 
4. Implement sorting feature to sort items by price, name ascending/descending

### CODE REVIEW

**Review by dvorjackz — CHANGES_REQUESTED** [review]
Taking an exam rn, but without looking at the code yet, could you show the quantity of each item? Also, make sure to merge in the latest main.

**Review by Saagar3690 — COMMENTED** [review]
### **Changes**
1. Added quantity field to each item (if quantity is 0 it displays sold out in bold and red as shown below)
2. Linked item to payment stubs page - when you press an items, it takes you to the payment page
   - passes in the item _id, amount (i.e price) which will be needed for Stripe API payment processing, and quantity (which for now is a hardcoded -1 since item count reduces in inventory)
3. Payment Stubs page now sends a request to the backend to update inventory (you can see below the count was 253 to 250 to 247)

### **In Action**
**Item with quantity > 0**
IMG_5428
**Sold Out**
IMG_5427
**In Action Video**
https://user-images.githubusercontent.com/30224372/102684118-b9987200-418a-11eb-9c1b-8b5c2c4fa6e8.mp4
**Response from backend after updating inventory**
Screen Shot 2020-12-18 at 11 31 21 PM
**Proof of quantity changing**
IMG_5431

### **TODO (by priority) **
1. Sale Statistics
2. Sort items by price, name, etc.
3. Anything else??

**Review by zhang-edward — APPROVED** [review]

File: src/containers/PaymentInfo.tsx [hunk 1]
```diff
@@ -8,20 +8,28 @@ import {
...
 }
 
 const screenWidth = Dimensions.get('window').width;
 const screenHeight = Dimensions.get('window').height;
 
-const PaymentInfo = ({ navigation }: PaymentInfoProps) => {
+const PaymentInfo = ({ navigation, route }: PaymentInfoProps) => {
+	/* OG Bot is BruinBear with id 5ff798be0390ab19822d21db for demo purposes (from local deb db),
+			but if you come from QR view, botId will be set from navigation params */
+	const [botId] = useState<string>(
```
Comment (zhang-edward) [inline src/containers/PaymentInfo.tsx:30]: I don't think `useState` is necessary here

```suggestion
	const botId = 
```

File: src/containers/PaymentInfo.tsx [hunk 2]
```diff
@@ -8,20 +8,28 @@ import {
...
+		route.params.botId ? route.params.botId : '5ff798be0390ab19822d21db'
```
Comment (zhang-edward) [inline src/containers/PaymentInfo.tsx:31]:
```suggestion
		route.params.botId ?? '5ff798be0390ab19822d21db'
```

**Review by zhang-edward — APPROVED** [review]
Left a few nitpick comments but otherwise looks fine

**Pull request merged**